src/webgpu/shader/validation/expression/access/vector.spec.ts

```diff
@@ -174,23 +174,189 @@ const kCases = {
   array_idx: { wgsl: 'let r : T = v[array<i32, 2>()];', ok: false },
 };
 
-g.test('vector')
-  .desc('Tests validation of vector indexed and swizzles')
+const kAbstractCases = {
+  // indexing with literal
+  literal_0: { wgsl: 'const r = V[0];', ok: true },
```

Comment:
Maybe as a followup, but it would be good to check the expression type (same thing for the concrete tests), by using an explicit type for `r` (`const r : T = V[0];`).

Unlike the concrete tests, the abstract tests would need to consider that `abstract-int` can convert to `abstract-float`, but the reverse is not true.

Reply:
I think I might be missing something about how abstracts in WGSL work.

I thought you couldn't explicitly type abstract int and float in WGSL, i.e. there is now way to write `const r: AbstractFloat`, it can only be inferred from the assignment.

And they are always constant (or override) experessions, so I cannot do something like `let r = 0; r = V[0];` to check that V[0] is an AbstractInt, because though the initializer literal is a AI, it will be converted on the assignement to r to be an i32, due to the use of `let`.

Reply:
You're right that you can't explicitly write an abstract numeric, and I see why what I wrote is confusing.

`T` would be a concrete type - `i32`, `u32`, `f32` etc. I was trying to point out that `abstract-int` could be used for any of these, but `abstract-float` would only work for `f32`.

Reply:
Maybe it would be better done as a second statement:

```wgsl
const C : T = r;
```

That would mean you still test that `r` gets the inferred abstract type.
You also wouldn't need to change all these individual cases, just add another statement underneath the templated case one.

Reply:
I understand now. I have filed https://github.com/gpuweb/cts/issues/3709 and will implement in a follow up CL.

Reply:
Currently we have nothing testing that the *inferred* type of `r` is as expected. The second assignment to a concrete type would validate that the inferred type is as expected.